qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/ModifyProcessInstanceTest.java

```diff
@@ -30,14 +34,65 @@ public class ModifyProcessInstanceTest {
 
   @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
   private String processId;
+  private String processId2;
   private long processDefinitionKey;
 
   @Before
   public void deploy() {
     processId = helper.getBpmnProcessId();
+    processId2 = helper.getBpmnProcessId();
     processDefinitionKey =
         CLIENT_RULE.deployProcess(
             Bpmn.createExecutableProcess(processId).startEvent().endEvent().done());
+    CLIENT_RULE.deployProcess(
+        Bpmn.createExecutableProcess(processId2)
+            .startEvent()
+            .userTask("A")
+            .parallelGateway()
+            .userTask("B")
+            .moveToLastGateway()
+            .userTask("C")
+            .endEvent()
+            .done());
+  }
+
+  @Test
+  public void shouldModifyExistingProcessInstance() {
+    // given
+    final var processInstance =
+        CLIENT_RULE
+            .getClient()
+            .newCreateInstanceCommand()
+            .bpmnProcessId(processId2)
+            .latestVersion()
+            .send()
+            .join();
+    final var processInstanceKey = processInstance.getProcessInstanceKey();
+
+    final var activatedUserTask =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .limit("A", ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .getFirst();
+
+    // when
+    final var command =
+        CLIENT_RULE
+            .getClient()
+            .newModifyProcessInstanceCommand(processInstanceKey)
+            .activateElement("B")
+            .withVariables(Map.of("foo", "bar"), "B")
+            .and()
+            .activateElement("C")
+            .withVariables(Map.of("fizz", "buzz"), "C")
+            .and()
+            .terminateElement(activatedUserTask.getKey())
+            .send();
+
+    // then
+    assertThatNoException()
+        .describedAs("Expect that modification command is not rejected")
+        .isThrownBy(command::join);
```

Comment:
:wrench: We could verify that the activation and termination instructions are applied by using the `RecordingExporter`.  

Currently, the test case would pass if the activate, terminate, or variable instructions are ignored (in the client, the gateway, or the broker).

Reply:
I've expanded the test case 👍